src/scripts.js

```diff
@@ -199,122 +184,130 @@ const displayRecipe = (id) => {
         ${ingredient.amount} ${ingredient.unit} ${ingredient.name} <span class="ingredients__message">You'll need xyz more of this.</span>
       </article>
     `;
-  })
+  });
   foundRecipe.instructions.forEach(instruction => {
     recipeInstructions.innerHTML += `
       <li>${instruction.number}. ${instruction.instruction}</li>
     `;
-  })
+  });
 }
 
-/* ~~~~~~~~~~~~~~~~~~~~~~~~~~~~Event Listeners~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/
+const splitInput = (input) => {
+  return input.value.split(' ');
+}
 
-allRecipesButton.addEventListener('click', displayAllRecipes);
-// recipeListView.addEventListener('click', displayRecipe);
-goButton.addEventListener('click', getSearchTerm);
-homeSelector.addEventListener('click', displayLanding);
-userSelector.addEventListener('click', displayPantry)
+const removeDuplicates = (arr) => {
+  return [...new Set(arr)];
+}
 
+const filterByTag = (searchTag, recipes) => {
+  return recipes.filter(recipe => {
+    if (recipe.tags.includes(searchTag.toLowerCase())) {
+      return recipe.id;
+    }
+  });
+}
 
-/* 📌 Katie's Ticket 📌 */
-const searchBarInput = document.querySelector('.search-bar');
-const searchButton = document.querySelector('.search-button');
-const searchError = document.querySelector('.search-error');
+const filterByIngredient = (searchIng, recipes) => {
+  return recipes.filter(recipe => {
+    return recipe.ingredients.find(ingredient => {
+      return ingredient.name.toLowerCase().includes(searchIng.toLowerCase());
+    });
+  });
+}
 
+const filterByName = (searchName, recipes) => {
+  return recipes.filter(recipe => {
+    return recipe.name.toLowerCase().includes(searchName.toLowerCase());
+  });
+}
 
-searchButton.addEventListener('click', function() {
-  search(searchBarInput);
-});
+// TODO make searchFor be an array of possible search terms ('appetizer', 'starter', etc.)
+// // which may blend well with the multi-selection that Nikki built
+const getTagToSearchFor = (choice) => {
+  let searchFor = '';
+  if (choice === 'appetizers') {
+    searchFor = 'appetizer';
+    // TODO also starter, hor d'oeuvre, hor d'oevres, antipasti, antipasto
+  } else if (choice === 'side-dishes') {
+    searchFor = 'side dish';
+  } else if (choice === 'main-courses') {
+    searchFor = 'main dish';
+    // TODO also main course, lunch, dinner
+  } else if (choice === 'desserts') {
+    searchFor = 'dessert';
+  } else {
+    searchFor = choice;
```

Comment:
Maybe we can refactor this to use a switch!!

Reply:
YES